js/runner.go

```diff
@@ -486,9 +486,10 @@ func (u *VU) runFn(
 
 	state.Samples <- u.Dialer.GetTrail(startTime, endTime, isFullIteration, stats.IntoSampleTags(&tags))
 
+	inVuStage := group.Name != stageSetup && group.Name != stageTeardown
```

Comment:
Looking at this I think ... this as well as the if below should be in `RunOnce` ... although .. I would've not have named it that way ... 
I find it strange that we check whether this is setup/teardown here ... and more importantly this will also probably break once we get more than just setup and teardown, but possibly per VU setup/teardown function (#785)  and per iteration init/teardown  and whatever we come up in the future

Reply:
Seem reasonable, as Runner.Setup, Runner.Teardown call runPart

Reply:
cc @na-- as he the one who added those code.